src/custom_nodes/model_zoo_intel_object_detection/model_zoo_intel_object_detection.cpp

```diff
@@ -14,38 +14,75 @@
 #include <iostream>
+#include <shared_mutex>
 #include <string>
 #include <vector>
 
 #include "../../custom_node_interface.h"
+#include "../common/buffersqueue.hpp"
+#include "../common/customNodeLibraryInternalManager.hpp"
 #include "opencv2/opencv.hpp"
 #include "utils.hpp"
 
+using CustomNodeLibraryInternalManager = ovms::custom_nodes_common::CustomNodeLibraryInternalManager;
+using BuffersQueue = ovms::custom_nodes_common::BuffersQueue;
+
 static constexpr const char* INPUT_IMAGE_TENSOR_NAME = "image";
 static constexpr const char* INPUT_DETECTION_TENSOR_NAME = "detection";
 static constexpr const char* OUTPUT_IMAGES_TENSOR_NAME = "images";
 static constexpr const char* OUTPUT_COORDINATES_TENSOR_NAME = "coordinates";
 static constexpr const char* OUTPUT_CONFIDENCES_TENSOR_NAME = "confidences";
+static constexpr const char* OUTPUT_TENSOR_NAME = "output";
+static constexpr const char* INPUT_TENSOR_INFO_NAME = "input_info";
+static constexpr const char* OUTPUT_TENSOR_INFO_NAME = "output_info";
+
+static constexpr const char* OUTPUT_IMAGES_DIMS_NAME = "images_dims";
+static constexpr const char* OUTPUT_COORDINATES_DIMS_NAME = "coordinates_dims";
+static constexpr const char* OUTPUT_CONFIDENCES_DIMS_NAME = "confidences_dims";
+static constexpr const char* INPUT_IMAGE_DIMS_NAME = "image_dims";
+static constexpr const char* INPUT_DETECTION_DIMS_NAME = "detection_dims";
+
+static constexpr const char* COORDINATES_INFO_DIMS_NAME = "coordinates_info_dims";
+static constexpr const char* IMAGES_INFO_DIMS_NAME = "images_info_dims";
+static constexpr const char* CONFIDENCES_INFO_DIMS_NAME = "confidences_info_dims";
+
+static constexpr const int STREAMS_LENGTH = 1;
+
+std::shared_mutex internalManagerLock;
+
+void cleanup(CustomNodeTensor& tensor, CustomNodeLibraryInternalManager* internalManager) {
+    release(tensor.data, internalManager);
+    release(tensor.dims, internalManager);
+}
+
+template <typename T>
+bool get_preallocated_buffer(CustomNodeLibraryInternalManager* internalManager, T** buffer, const char* buffersQueueName, uint64_t byte_size) {
+    *buffer = static_cast<T*>(internalManager->getBuffersQueue(buffersQueueName)->getBuffer());
+    if (*buffer == nullptr) {
+        *buffer = (T*)malloc(byte_size);
+        if (*buffer == nullptr) {
+            std::cout << "allocation for buffer: " << buffersQueueName << "FAILED" << std::endl;
+            return false;
+        }
+    }
+    return true;
+}
 
-bool copy_images_into_output(struct CustomNodeTensor* output, const std::vector<cv::Rect>& boxes, const cv::Mat& originalImage, int targetImageHeight, int targetImageWidth, const std::string& targetImageLayout, bool convertToGrayScale) {
+bool copy_images_into_output(struct CustomNodeTensor* output, const std::vector<cv::Rect>& boxes, const cv::Mat& originalImage, int targetImageHeight, int targetImageWidth, const std::string& targetImageLayout, bool convertToGrayScale, void* customNodeLibraryInternalManager) {
```

Comment:
Here you do not need to have internal manager as `void*`. You can use `CustomNodeLibraryInternalManager*` type since this is not API function implementation. This will reduce the code length and make it easier to read.